### ISSUE TITLE
Could not acquire lock ovs_lock_ovs_dataversion during set_domain

### ISSUE THREAD

**Issue by kinvaris** [openvstorage/framework#1090]
While performing our regular CI tests, we saw this problem:

```
Oct 27 11:03:21 ovs-node01-1604 gunicorn[10704]: 2016-10-27 11:03:21 28700 +0200 - ovs-node01-1604 - 10737/139678508535312 - log/api - 4 - INFO - [storagerouters.set_domains] - 28c78292-f260-498f-939c-30f65115ce93 - [] - {"pk": "1ac825fd-dc74-444a-8cbf-d04e9110994a"} - {"cookies": {}, "meta": {"HTTP_AUTHORIZATION": "Bearer ElcdH?|Gn3x=oSQf{DDsNdR2X>i.2zow\"krb7\"#?u]?RKQOJz3KYFT'x2s>eus+v", "wsgi.multiprocess": "True", "HTTP_X_FORWARDED_SSL": "on", "SERVER_SOFTWARE": "gunicorn/19.4.5", "SCRIPT_NAME": "/api", "REQUEST_METHOD": "POST", "PATH_INFO": "/storagerouters/1ac825fd-dc74-444a-8cbf-d04e9110994a/set_domains/", "SERVER_PROTOCOL": "HTTP/1.0", "QUERY_STRING": "", "HTTP_X_REAL_IP": "10.100.199.171", "CONTENT_LENGTH": "165", "HTTP_USER_AGENT": "python-requests/2.9.1", "HTTP_CONNECTION": "close", "SERVER_NAME": "127.0.0.1", "REMOTE_PORT": "39386", "wsgi.url_scheme": "https", "SERVER_PORT": "8002", "HTTP_ACCEPT": "application/json; version=*", "wsgi.input": "<gunicorn.http.body.Body object at 0x7f09572b4bd0>", "HTTP_HOST": "10.100.199.171", "wsgi.multithread": "True", "HTTP_X_SCHEME": "https", "wsgi.version": "(1, 0)", "RAW_URI": "/storagerouters/1ac825fd-dc74-444a-8cbf-d04e9110994a/set_domains/", "wsgi.run_once": "False", "wsgi.errors": "<gunicorn.http.wsgi.WSGIErrorsWrapper object at 0x7f09572b46d0>", "REMOTE_ADDR": "127.0.0.1", "gunicorn.socket": "<socket fileno=15 sock=127.0.0.1:8002 peer=127.0.0.1:39386>", "CONTENT_TYPE": "application/json", "wsgi.file_wrapper": "<class 'gunicorn.http.wsgi.FileWrapper'>", "CSRF_COOKIE": "COhEN6Csx4rgbhYqfq6i97UMCqW2ISjW", "HTTP_ACCEPT_ENCODING": "gzip, deflate"}, "request": {}}
Oct 27 11:03:51 ovs-node01-1604 gunicorn[10704]: 2016-10-27 11:03:51 30100 +0200 - ovs-node01-1604 - 10737/139678508535312 - extensions/volatile mutex - 5 - ERROR - Lock for ovs_lock_ovs_dataversion_storagerouter_1ac825fd-dc74-444a-8cbf-d04e9110994a could not be acquired. 30.0048961639 sec > 30 sec
Oct 27 11:03:51 ovs-node01-1604 gunicorn[10704]: 2016-10-27 11:03:51 30300 +0200 - ovs-node01-1604 - 10737/139678508535312 - api/middleware - 6 - ERROR - An unhandled exception occurred: Could not acquire lock ovs_lock_ovs_dataversion_storagerouter_1ac825fd-dc74-444a-8cbf-d04e9110994a
Oct 27 11:03:51 ovs-node01-1604 gunicorn[10704]: Traceback (most recent call last):
Oct 27 11:03:51 ovs-node01-1604 gunicorn[10704]:   File "/usr/lib/python2.7/dist-packages/django/core/handlers/base.py", line 132, in get_response
Oct 27 11:03:51 ovs-node01-1604 gunicorn[10704]:     response = wrapped_callback(request, *callback_args, **callback_kwargs)
Oct 27 11:03:51 ovs-node01-1604 gunicorn[10704]:   File "/usr/lib/python2.7/dist-packages/rest_framework/viewsets.py", line 78, in view
Oct 27 11:03:51 ovs-node01-1604 gunicorn[10704]:     return self.dispatch(request, *args, **kwargs)
Oct 27 11:03:51 ovs-node01-1604 gunicorn[10704]:   File "/usr/lib/python2.7/dist-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
Oct 27 11:03:51 ovs-node01-1604 gunicorn[10704]:     return view_func(*args, **kwargs)
Oct 27 11:03:51 ovs-node01-1604 gunicorn[10704]:   File "/usr/lib/python2.7/dist-packages/rest_framework/views.py", line 399, in dispatch
Oct 27 11:03:51 ovs-node01-1604 gunicorn[10704]:     response = self.handle_exception(exc)
Oct 27 11:03:51 ovs-node01-1604 gunicorn[10704]:   File "/usr/lib/python2.7/dist-packages/rest_framework/views.py", line 396, in dispatch
Oct 27 11:03:51 ovs-node01-1604 gunicorn[10704]:     response = handler(request, *args, **kwargs)
Oct 27 11:03:51 ovs-node01-1604 gunicorn[10704]:   File "/opt/OpenvStorage/webapps/api/backend/decorators.py", line 461, in new_function
Oct 27 11:03:51 ovs-node01-1604 gunicorn[10704]:     return_value = f(*args, **kwargs)
Oct 27 11:03:51 ovs-node01-1604 gunicorn[10704]:   File "/opt/OpenvStorage/webapps/api/backend/decorators.py", line 79, in new_function
Oct 27 11:03:51 ovs-node01-1604 gunicorn[10704]:     return f(*args, **kw)
Oct 27 11:03:51 ovs-node01-1604 gunicorn[10704]:   File "/opt/OpenvStorage/webapps/api/backend/decorators.py", line 365, in new_function
Oct 27 11:03:51 ovs-node01-1604 gunicorn[10704]:     result = f(*args, **kwargs)
Oct 27 11:03:51 ovs-node01-1604 gunicorn[10704]:   File "/opt/OpenvStorage/webapps/api/backend/decorators.py", line 151, in new_function
Oct 27 11:03:51 ovs-node01-1604 gunicorn[10704]:     instance = object_type(kwargs['pk'])
Oct 27 11:03:51 ovs-node01-1604 gunicorn[10704]:   File "/opt/OpenvStorage/ovs/dal/dataobject.py", line 274, in __init__
Oct 27 11:03:51 ovs-node01-1604 gunicorn[10704]:     self._mutex_version.acquire(30)
Oct 27 11:03:51 ovs-node01-1604 gunicorn[10704]:   File "/opt/OpenvStorage/ovs/extensions/generic/volatilemutex.py", line 77, in acquire
Oct 27 11:03:51 ovs-node01-1604 gunicorn[10704]:     raise NoLockAvailableException('Could not acquire lock {0}'.format(self.key()))
Oct 27 11:03:51 ovs-node01-1604 gunicorn[10704]: NoLockAvailableException: Could not acquire lock ovs_lock_ovs_dataversion_storagerouter_1ac825fd-dc74-444a-8cbf-d04e9110994a
```

**Comment by khenderick** [comment]
Fixed by #1106, packaged in openvstorage-2.7.4-rev.4230.a30de27

**Comment by pploegaert** [comment]
Validated manually:
- code present
- error is logged
- NoLockAvailableException raised
